src/nomad_analysis/schema.py

```diff
@@ -201,13 +202,19 @@ def set_jupyter_notebook_name(
             archive (EntryArchive): The archive containing the section.
             logger (BoundLogger): A structlog logger.
         """
-        file_name = (
-            self.name.replace(' ', '_')
-            + f'_{self.analysis_type.lower()}_notebook.ipynb'
-        )
-        if not archive.m_context.raw_path_exists(self.notebook):
+        if self.name:
+            file_name = (
+                self.name.replace(' ', '_')
+                + f'_{self.analysis_type.lower()}_notebook.ipynb'
+            )
+        else:
+            file_name = 'untitled.ipynb'
+
+        if self.notebook is None:
             self.notebook = file_name
-        elif not self.notebook == file_name:
+            return
+
+        if self.notebook != file_name:
             raw_path = archive.m_context.raw_path()
```

Comment:
Question: is this now working with your changes in Context? 🙂

Reply:
Yes, it does now. The issue I was facing was that with the ClientContext, `self.name` which is a quantity defined in the  class `BaseSection` was None. The previous code wasn't expecting a None value. 

In future too, we will have to write code that is compatible with both ClientContext and ServerContext. We will see how challenging it becomes (or not at all).